src/harp-serial/harp/serial/device.py

```diff
@@ -0,0 +1,1347 @@
+from __future__ import annotations  # for type hints (PEP 563)
+
+import logging
+import queue
+from enum import Enum
+from io import BufferedWriter
+from pathlib import Path
+from typing import Optional
+
+import serial
+from harp.protocol import (
+    ClockConfig,
+    CommonRegisters,
+    MessageType,
+    OperationCtrl,
+    OperationMode,
+    PayloadType,
+    ResetMode,
+)
+from harp.protocol.exceptions import HarpException, HarpTimeoutException
+from harp.protocol.messages import HarpMessage
+from harp.serial.harp_serial import HarpSerial
+
+
+class TimeoutStrategy(Enum):
+    """
+    Strategy to handle timeouts when waiting for a reply from the device.
+
+    Attributes
+    ----------
+    RAISE : str
+        Raise HarpTimeoutException
+    RETURN_NONE : str
+        Return None
+    LOG_AND_RAISE : str
+        Log the timeout and raise HarpTimeoutException
+    LOG_AND_NONE : str
+        Log the timeout and return None
+    """
+
+    RAISE = "raise"  # Raise HarpTimeoutException
+    RETURN_NONE = "return_none"  # Return None
+    LOG_AND_RAISE = "log_and_raise"
+    LOG_AND_NONE = "log_and_none"
+
+
+class Device:
+    """
+    The `Device` class provides the interface for interacting with Harp devices. This implementation of the Harp device was based on the official documentation available on the [harp-tech website](https://harp-tech.org/protocol/Device.html).
+
+    Attributes
+    ----------
+    WHO_AM_I : int
+        The device ID number. A list of devices can be found [here](https://github.com/harp-tech/protocol/blob/main/whoami.md)
+    HW_VERSION_H : int
+        The major hardware version
+    HW_VERSION_L : int
+        The minor hardware version
+    ASSEMBLY_VERSION : int
+        The version of the assembled components
+    HARP_VERSION_H : int
+        The major Harp core version
+    HARP_VERSION_L : int
+        The minor Harp core version
+    FIRMWARE_VERSION_H : int
+        The major firmware version
+    FIRMWARE_VERSION_L : int
+        The minor firmware version
+    DEVICE_NAME : str
+        The device name stored in the Harp device
+    SERIAL_NUMBER : int, optional
+        The serial number of the device
+    """
+
+    WHO_AM_I: int
+    HW_VERSION_H: int
+    HW_VERSION_L: int
+    ASSEMBLY_VERSION: int
+    CORE_VERSION_H: int
+    CORE_VERSION_L: int
+    FIRMWARE_VERSION_H: int
+    FIRMWARE_VERSION_L: int
+    DEVICE_NAME: str
+    SERIAL_NUMBER: int
+    CLOCK_CONFIG: int
+    TIMESTAMP_OFFSET: int
+
+    _ser: HarpSerial
+    _dump_file_path: Optional[Path]
```

Comment:
Is this actually being used anywhere?

I like the idea of having a logger that makes everything log in the standard format. But this should be decoupled from the device class.
Maybe lets remove this implementation for now?